python/rmm/_lib/device_buffer.pyx

```diff
@@ -9,18 +9,25 @@ from rmm._lib.lib cimport (cudaError_t, cudaSuccess,
 
 cdef class DeviceBuffer:
 
-    def __cinit__(self, *, ptr=None, size=None):
+    def __cinit__(self, *, ptr=None, size=None, stream=None):
         if size is None:
             size = 0
 
+        if stream is None:
+            stream = 0
+
         cdef void * data
         if ptr is None:
-            self.c_obj.reset(new device_buffer(<size_t>size))
+            self.c_obj.reset(
+                new device_buffer(<size_t>size, <cudaStream_t>stream)
```

Comment:
I think we need a doublecast if someone passes us a python integer here.
```suggestion
                new device_buffer(<size_t>size, <cudaStream_t><uintptr_t>stream)
```

Reply:
Good catch. Thanks!